### PR TITLE
[STRY/BE] REST API: Bullets

### LINKED ISSUE #2: [STRY/BE] REST API: Bullets
**Briefly Describe the Feature-Set for the Story**
Implement REST API for Bullets.

**Layout acceptance criteria**
CRUD for bullets. All request types function in Postman.

**Briefly Describe the Feature-Set for the Story**
Handle CRUD requests sent to the express server

**Layout acceptance criteria**
Contains all fields for bullets and links properly to other tables. 

Requires the following fields contained in bullet:
- Primary key
- User
- Field saying it is a bullet
- Signifier (importance)
- Type (Events, tasks, notes)
- Body content
- Date

**Additional context (optional)**
Entangled and partially blocked by Daily and Monthly log
Blocked by agnostic updating function

### PR DESCRIPTION
**Pull request by dino-inc**
Adds CRUD for bullets, so that they can be created, updated, and deleted
Closes #2

### CODE REVIEW

File: source/routes/update/bullet.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
+const PouchDB = require('pouchdb');
+PouchDB.plugin(require("pouchdb-find"));
+const db = new PouchDB('db');
```
Comment (b2wu) [inline source/routes/update/bullet.js:3]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/cse110-sp21-group13/cse110-sp21-group13/pullRequest?prid=7402063)
```suggestion
const db = new PouchDB("db");
```

File: source/routes/read/bullet.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
+const PouchDB = require('pouchdb');
+PouchDB.plugin(require("pouchdb-find"));
+const db = new PouchDB('db');
```
Comment (b2wu) [inline source/routes/read/bullet.js:3]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/cse110-sp21-group13/cse110-sp21-group13/pullRequest?prid=7402063)
```suggestion
const db = new PouchDB("db");
```

File: source/routes/delete/bullet.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
+const PouchDB = require('pouchdb');
+PouchDB.plugin(require("pouchdb-find"));
+const db = new PouchDB('db');
```
Comment (b2wu) [inline source/routes/delete/bullet.js:3]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/cse110-sp21-group13/cse110-sp21-group13/pullRequest?prid=7402063)
```suggestion
const db = new PouchDB("db");
```

File: source/routes/create/bullet.js [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+const db = new PouchDB("db");
+
+// Send in json form of the bullet and it gets sent to the database
+module.exports = {
+    '/create/bullet': {
+        methods: ['post'],
+        fn: function(req, res, next) {
+            // Check if every field exists, if not, throw error
+            let requiredFields = ["parentDocId", "user", "signifier", "bulletType", "content", "date"];
+            requiredFields.forEach((jsonField, index)=>{
```
Comment (b2wu) [inline source/routes/create/bullet.js:12]: Codacy has a fix for the issue: [Missing space before =>.](https://app.codacy.com/gh/cse110-sp21-group13/cse110-sp21-group13/pullRequest?prid=7402063)
```suggestion
            requiredFields.forEach((jsonField, index) =>{
```

File: source/routes/create/bullet.js [hunk 5]
```diff
@@ -0,0 +1,58 @@
+const PouchDB = require('pouchdb');
+PouchDB.plugin(require("pouchdb-find"));
+const db = new PouchDB("db");
+
+// Send in json form of the bullet and it gets sent to the database
+module.exports = {
+    '/create/bullet': {
+        methods: ['post'],
```
Comment (b2wu) [inline source/routes/create/bullet.js:8]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/cse110-sp21-group13/cse110-sp21-group13/pullRequest?prid=7402063)
```suggestion
        methods: ["post"],
```

File: source/routes/update/bullet.js [hunk 6]
```diff
@@ -0,0 +1,48 @@
+const PouchDB = require('pouchdb');
```
Comment (b2wu) [inline source/routes/update/bullet.js:1]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/cse110-sp21-group13/cse110-sp21-group13/pullRequest?prid=7402063)
```suggestion
const PouchDB = require("pouchdb");
```

File: source/routes/create/bullet.js [hunk 7]
```diff
@@ -0,0 +1,58 @@
+const PouchDB = require('pouchdb');
+PouchDB.plugin(require("pouchdb-find"));
+const db = new PouchDB("db");
+
+// Send in json form of the bullet and it gets sent to the database
+module.exports = {
+    '/create/bullet': {
```
Comment (b2wu) [inline source/routes/create/bullet.js:7]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/cse110-sp21-group13/cse110-sp21-group13/pullRequest?prid=7402063)
```suggestion
    "/create/bullet": {
```

File: source/routes/delete/bullet.js [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+
+// Send in a valid json containing just the id and it will be expunged from the database
+module.exports = {
+    "/delete/bullet": {
+        methods: ['delete'],
```
Comment (b2wu) [inline source/routes/delete/bullet.js:8]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/cse110-sp21-group13/cse110-sp21-group13/pullRequest?prid=7402063)
```suggestion
        methods: ["delete"],
```

File: source/routes/create/bullet.js [hunk 9]
```diff
@@ -0,0 +1,58 @@
+const PouchDB = require('pouchdb');
```
Comment (b2wu) [inline source/routes/create/bullet.js:1]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/cse110-sp21-group13/cse110-sp21-group13/pullRequest?prid=7402063)
```suggestion
const PouchDB = require("pouchdb");
```

File: source/routes/read/bullet.js (hunk 2 above)
Comment (b2wu) [inline source/routes/read/bullet.js:3]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/cse110-sp21-group13/cse110-sp21-group13/pullRequest?prid=7402063)
```suggestion
const db = new PouchDB("db");
```

File: source/routes/read/bullet.js [hunk 10]
```diff
@@ -0,0 +1,18 @@
+const PouchDB = require('pouchdb');
```
Comment (b2wu) [inline source/routes/read/bullet.js:1]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/cse110-sp21-group13/cse110-sp21-group13/pullRequest?prid=7402063)
```suggestion
const PouchDB = require("pouchdb");
```

File: source/routes/update/bullet.js [hunk 11]
```diff
@@ -0,0 +1,41 @@
...
+PouchDB.plugin(require("pouchdb-find"));
+const db = new PouchDB('db');
```
Comment (b2wu) [inline source/routes/update/bullet.js:3]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/cse110-sp21-group13/cse110-sp21-group13/pullRequest?prid=7402063)
```suggestion
const db = new PouchDB("db");
```

File: source/routes/delete/bullet.js (hunk 3 above)
Comment (b2wu) [inline source/routes/delete/bullet.js:3]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/cse110-sp21-group13/cse110-sp21-group13/pullRequest?prid=7402063)
```suggestion
const db = new PouchDB("db");
```

File: source/routes/create/bullet.js [hunk 12]
```diff
@@ -0,0 +1,58 @@
...
+const db = new PouchDB("db");
+
+// Send in json form of the bullet and it gets sent to the database
+module.exports = {
+    "/create/bullet": {
+        methods: ["post"],
+        fn: function(req, res, next) {
+            // Check if every field exists, if not, throw error
+            let requiredFields = ["parentDocId", "user", "signifier", "bulletType", "content", "date"];
+            requiredFields.forEach((jsonField, index) =>{
```
Comment (b2wu) [inline source/routes/create/bullet.js:12]: Codacy has a fix for the issue: [Missing space before =>.](https://app.codacy.com/gh/cse110-sp21-group13/cse110-sp21-group13/pullRequest?prid=7402063)
```suggestion
            requiredFields.forEach((jsonField, index) =>{
```

**Review by zhl024 — APPROVED** [review]
The code seems pretty neat and no need to change.

**Review by ryankeegan — APPROVED** [review]
Looks good to me. Fingers crossed on the merge.

**Pull request merged**